css/BEM-style.md

```diff
@@ -0,0 +1,223 @@
+
+- **"My CSS styles never have conflicts since I utilize ID's most often to style my elements"**   
+Using ID's can definitely help you avoid any kind of specificity conflicts, since they have a priority over classes and tags, but that would mean, your styles are not reusable and you don't follow the DRY( Dont'Repeat Yourself) principle. New styles need to be written for every element even if some of them share a few common feature (Eg: buttons on a page, navlinks etc).
+
+### Guided Practice
+We'll make use of the BEM method to create few buttons and a hero-box .
+
+> Note: You must have prior knowledge of CSS and HTML to understand what BEM is and how to make use of it.
+
+**1.Understanding what BEM is?:** 
+
+BEM is a naming methodology/principle which stand for "Block", "Element", "Modifier" to describe and differentiate, the different components of your webpage in a way that is easy to understand,maintain and scale as and when required.
+In BEM methodology the important point to note is that each and every HTML element will be accessed using a class only, we don't use ID's or tags to style them, to keep specificity uniform across the entire project i.e. (0,1,0) : (ID's, Classes, Tags).
+
+- The **'Block'** are the standalone components in a webpage that are meaningful on its own.
+ 
+  >Example includes: navbar,header,footer,hero-boxes, buttons etc.
+
+-  The **'Element'** are the components that have no standalone meaning, but when tied to a block, represents a part of the block. It expresses a semantic meaning only when tied to a block.  
+   
+   >Example includes: nav-links,footer-links, headings and sub headings inside heroboxes etc.
+
+-  The **'Modifier'** are like flags that differentiate blocks or elements inside block in terms of its appreance.
+   
+   >Example includes: buttons which are muted, navlinks which are active, disabled etc. 
+
+
+**2.Understanding how BEM is written (The Syntax) ?:** 
+
+To understand this, let us consider that we are making two different buttons: one which is in active state and one which is in ghosted state.
+Then the HTML makrup with the classes will be:
+```
+<a href="#"class="btn btn--active">I know!</a>
+<a href="#"class="btn btn--ghosted">I surely know!</a>
+```
+
+Notice the pattern in naming of the classes, ***.btn*** is the block since it has a standalone meaning, while ***.btn--active*** and ***.btn--ghosted*** are modifiers which do not have a standalone meaning but when sematically attached to it's block (the button) describe the modification (flags) performed on the button (i.e) active modifier/flag and ghosted modifier/flag.  
+The modifiers are named by appeneding `--` to the element which you wish to modify.
+
+Now these buttons will be a part of hero-box or any call to action section. We'll  write the markup that describes that, with the necessary classes.
+ 
+```
+<div class="hero-box__buttons">
+    <a href="#"class="btn btn--active">I know!</a>
```

Comment:
I don't really understand what this example is demonstrating. Can you provide more realistic example with context?

Why would "I know!" be active and "I surely know!" be *ghosted* (as @alodahl mentioned, not sure what that means)?

Also, why would we have links inside a button group?